scripts/verify-last-flag-matches-upstream.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+#
+# Copyright 2021 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# This script verifies the LastFlag response flag in istio/proxy matches the one in envoyproxy/envoy 
+# and fails with a non-zero exit code if they differ.
+
+set -x
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+WORKSPACE=${ROOT}/WORKSPACE
+PATH_LASTFLAG_SPEC_DOWNSTREAM="${ROOT}/extensions/common/util.cc"
+PATH_LASTFLAG_SPEC_UPSTREAM="stream_info.h"
+ENVOY_PATH_LASTFLAG_SPEC="envoy/stream_info/stream_info.h"
+
+ENVOY_ORG="$(grep -Pom1 "^ENVOY_ORG = \"\K[a-zA-Z-]+" "${WORKSPACE}")"
+ENVOY_REPO="$(grep -Pom1 "^ENVOY_REPO = \"\K[a-zA-Z-]+" "${WORKSPACE}")"
+ENVOY_LATEST_SHA="$(git ls-remote https://github.com/"${ENVOY_ORG}"/"${ENVOY_REPO}" "main" | awk '{ print $1}')"
+curl -sSL "https://raw.githubusercontent.com/${ENVOY_ORG}/${ENVOY_REPO}/${ENVOY_LATEST_SHA}/${ENVOY_PATH_LASTFLAG_SPEC}" > "${PATH_LASTFLAG_SPEC_UPSTREAM}"
+
+# Extract the LastFlag specification from both sources and trim all white spaces.
+
+if grep -q "^\s*LastFlag\s*=.*$" "${PATH_LASTFLAG_SPEC_DOWNSTREAM}"
+then
+    DOWNSTREAM_LASTFLAG=$(grep -m1 "^\s*LastFlag\s*=.*$" "${PATH_LASTFLAG_SPEC_DOWNSTREAM}"|tr -d '[:space:]')
+else
+    echo "istio/proxy: LastFlag not specified in ${PATH_LASTFLAG_SPEC_DOWNSTREAM}"
+    rm "${PATH_LASTFLAG_SPEC_UPSTREAM}"
```

Comment:
nit: you can use trap on exit, to avoid adding rm on every branch: `trap 'rm -rf ${PATH_LASTFLAG_SPEC_UPSTREAM}' EXIT`